android/quest/src/main/java/org/smartregister/fhircore/quest/util/SearchExtension.kt

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright 2021 Ona Systems, Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.smartregister.fhircore.quest.util
+
+import ca.uhn.fhir.rest.gclient.ReferenceClientParam
+import com.google.android.fhir.search.Search
+import org.hl7.fhir.r4.model.ResourceType
+
+fun Search.filterByPractitioner(reference: ReferenceClientParam, practitionerId: String) {
```

Comment:
We can generalise Search.filterByPatient method of engine Search extension to Search.filterByResourceType by passing ResourceType name as a parameter and use that here as well .(https://github.com/opensrp/fhircore/blob/main/android/engine/src/main/java/org/smartregister/fhircore/engine/util/extension/SearchExtension.kt)

Reply:
Yeah sure, I think it's better that way